packages/ui/src/hooks/use-social-sign-in-listener.ts

```diff
@@ -1,51 +1,72 @@
+import type { RequestErrorBody } from '@logto/schemas';
 import { SignInMode } from '@logto/schemas';
 import { useEffect, useCallback, useContext, useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useParams, useNavigate } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import { validate } from 'superstruct';
```

Comment:
nit: Are we still using superstruct in mainflow UI?

Reply:
yes